service/auth_client/auth_client.go

```diff
@@ -60,6 +66,25 @@ func (r *authClient) AuthnWithDid(did string, nonce string, enc []byte) error {
 	return nil
 }
 
+func (r *authClient) CreateDidRegistrationChallenge(auth_token string) (*DidRegistrationChallenge, error) {
```

Comment:
Not sure how this is going to work? When you start, it is a stream. It is waiting for 2 responses. One is a challenge, then wait for another one. If you just wrap it up in this call, then the stream will just timeout.

Reply:
Done